src/onemod/stage/base.py

```diff
@@ -149,52 +188,88 @@ def to_json(self, filepath: Path | str | None = None) -> None:
         filepath = filepath or self.directory / (self.name + ".json")
         with open(filepath, "w") as f:
             f.write(self.model_dump_json(indent=4, exclude_none=True))
+            
+    def to_dict(self) -> dict:
+        """Convert stage to dictionary representaion."""
+        return {
+            "name": self.name,
+            "type": self.type,
+            "config": self.config.model_dump(),
+            "input": self.input.to_dict(),
+            "output": self.output.to_dict(),
+            "dependencies": self.dependencies,
+        }
 
-    @classmethod
+    @validate_call
     def evaluate(
-        cls,
-        filepath: Path | str,
-        stage_name: str | None = None,
+        self,
+        config: Path | str | None = None,
         from_pipeline: bool = False,
-        method: str = "run",
+        method: Literal["run", "fit", "predict", "collect"] = "run",
+        backend: Literal["local", "jobmon"] = "local",
         *args,
         **kwargs,
     ) -> None:
         """Evaluate stage method.
 
         Parameters
         ----------
-        filepath : Path or str
-            Path to config file.
-        stage_name : str or None, optional
-            Stage name, required if `from_pipeline` is True.
+        config : Path, str, or None, optional
+            Path to config file. Required if `backend` is 'jobmon'.
             Default is None.
         from_pipeline : bool, optional
-            Whether `filepath` is a pipeline or stage config file.
+            Whether `config` is a pipeline or stage config file.
             Default is False.
         method : str, optional
-            Name of stage method to evaluate. Default is 'run'.
-
-        Notes
-        -----
-        This class method is designed to be called from the command
-        line. It creates a stage instance from the config file, then
-        calls the stage method.
+            Name of method to evaluate. Default is 'run'.
+        backend : str, optional
+            Whether to evaluate the method locally or with Jobmon.
+            Default is 'local'.
 
         """
-        stage = cls.from_json(filepath, stage_name, from_pipeline)
-        if method in stage.skip_if:
-            raise AttributeError(f"{stage.name} skips the '{method}' method")
+        if method in self.skip_if:
+            raise AttributeError(f"{self.name} skips the '{method}' method")
+        if backend == "jobmon":
+            raise NotImplementedError("Jobmon backend not implemented")
         try:
-            stage.__getattribute__(method)(*args, **kwargs)
+            self.__getattribute__(method)()
```

Comment:
I think we still need to pass arguments (could be subset_id, param_id), I accidentally remove them but added them back after trying to run things